server/handler.go

```diff
@@ -144,3 +146,38 @@ func (h *Handler) ProcessOptions() http.HandlerFunc {
 		writeJSON(http.StatusOK, struct{}{}, w)
 	}
 }
+
+func (h *Handler) SetupPubsub(cfg *config.Config) {
+	ctx := context.Background()
+	client, err := pubsub.NewClient(ctx, cfg.Pubsub.Project)
+	if err != nil {
+		logrus.WithError(err).Fatal("unable to set up pubsub client")
+	}
+
+	topic := client.Topic(cfg.Pubsub.Topic)
+	exists, err := topic.Exists(ctx)
+	if err != nil {
+		logrus.WithError(err).Fatal("unable to configure topic")
+	}
+	if !exists {
+		e := fmt.Errorf("topic %v does not exist in %v", cfg.Pubsub.Topic, cfg.Pubsub.Project)
+		logrus.WithError(e).Fatal("topic does not exist")
+	}
+	sub := client.Subscription(cfg.Pubsub.Subscription)
+	err = sub.Receive(ctx, func(ctx context.Context, msg *pubsub.Message) {
+		var data map[string]interface{}
+		defer msg.Ack()
+		if err := json.Unmarshal(msg.Data, &data); err != nil {
+			logrus.WithError(err).Error("unable to process message")
+			return
+		}
+
+		go func(buildUrl, buildID string) {
+			crawler := h.crawlerCreator(h.database, h.config)
+			crawler.CrawlJenkins(buildUrl, buildID)
+		}(data["buildUrl"].(string), data["correlationId"].(string))
```

Comment:
What about consuming an event following https://github.com/cloudevents/spec/blob/master/spec.md#design-goals here ? :)

Reply:
I actually considered it, but I also wanted it to fit in with our existing messages (and potentially making the keys configurable)

Reply:
but you have a point, maybe I'm optimizing for the wrong thing